ui/org.eclipse.pde.ui/src/org/eclipse/pde/ui/launcher/PluginsTab.java

```diff
@@ -188,7 +202,9 @@ public void performApply(ILaunchConfigurationWorkingCopy configuration) {
 		int index = fSelectionCombo.getSelectionIndex();
 		configuration.setAttribute(IPDELauncherConstants.USE_DEFAULT, index == DEFAULT_SELECTION);
 		configuration.setAttribute(IPDELauncherConstants.USE_CUSTOM_FEATURES, index == FEATURE_SELECTION);
-		fBlock.performApply(configuration);
+		if (fActivated) {
```

Comment:
> Ready for review. The NPE was because of a premature invocation of `AbstractPluginBlock.performApply`. Skipping that invocation until the block has been created (_i.e._ until **after** `createControl` has been called) fixes the issue and has no side-effects.

I think preventing the block's performApply method here already is better since we don't need to maintain the `isActivated` state in another place too.